### PR TITLE
[DOC] Enhanced RDoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Method +install+ is the first for which I noticed Pathname in the tests.  Some methods previously enhanced may need more about Pathnames;  I've made a note.

### CODE REVIEW

File: lib/fileutils.rb [hunk 1]
```diff
@@ -1324,8 +1340,8 @@ def compare_file(a, b)
   module_function :identical?
   module_function :cmp
 
-  #
-  # Returns true if the contents of a stream +a+ and +b+ are identical.
+  # Returns +true+ if the contents of streams +a+ and a file +b+ are identical,
```
Comment (peterzhu2118) [inline lib/fileutils.rb:1343]: Ditto

File: lib/fileutils.rb [hunk 2]
```diff
@@ -1342,13 +1358,52 @@ def compare_stream(a, b)
   end
   module_function :compare_stream
 
+  # Copies the file entry at path +src+ to the entry at path +dest+;
+  # each of +src+ and +dest+ may be a string or a
+  # {Pathnames}[https://docs.ruby-lang.org/en/master/Pathname.html].
```
Comment (peterzhu2118) [inline lib/fileutils.rb:1363]:
```suggestion
  # {Pathname}[https://docs.ruby-lang.org/en/master/Pathname.html].
```

File: lib/fileutils.rb [hunk 3]
```diff
@@ -1342,13 +1358,52 @@ def compare_stream(a, b)
...
+  #   File.read('dest2/src2.txt') # => "aaa\n"
+  #
+  # Keyword arguments:
+  #
+  # - <tt>group: <i>group</i></tt> - changes the group.
+  # - <tt>mode: <i>permissions</i></tt> - changes the permissions.
+  # - <tt>noop: true</tt> - does not remove entries; returns +nil+.
+  # - <tt>owner: <i>owner</i></tt> - changes the owner.
+  # - <tt>preserve: true</tt> - preserve timestamps.
+  # - <tt>verbose: true</tt> - prints an equivalent command:
```
Comment (peterzhu2118) [inline lib/fileutils.rb:1396]: The format of the keyword arguments is inconsistent here. Some (`noop`, `preserve`, `verbose`) show their default values, which others (`group`, `mode`, `owner`) just repeat the variable name as the value. I think we should just show their default values.

File: lib/fileutils.rb [hunk 4]
```diff
@@ -1177,6 +1177,11 @@ def rm_rf(list, noop: nil, verbose: nil, secure: nil)
   # Avoids a local vulnerability that can exist in certain circumstances;
   # see {Avoiding the TOCTTOU Vulnerability}[rdoc-ref:FileUtils@Avoiding+the+TOCTTOU+Vulnerability].
   #
+  # Keyword arguments:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed (three places).

File: lib/fileutils.rb [hunk 5]
```diff
@@ -1283,9 +1292,14 @@ def remove_entry(path, force = false)
   end
   module_function :remove_entry
 
+  # Removes the file entry given by +path+,
+  # which should be the entry for a regular file, a symbolic link,
+  # or a directory.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb [hunk 6]
```diff
@@ -1294,20 +1308,22 @@ def remove_file(path, force = false)
...
     remove_entry path, force   # FIXME?? check if it is a directory
   end
   module_function :remove_dir
 
-  #
-  # Returns true if the contents of a file +a+ and a file +b+ are identical.
-  #
-  #   FileUtils.compare_file('somefile', 'somefile')       #=> true
-  #   FileUtils.compare_file('/dev/null', '/dev/urandom')  #=> false
+  # Returns +true+ if the contents of files +a+ and a file +b+ are identical,
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb (hunk 1 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb (hunk 2 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb (hunk 3 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: The first three you mention show non-default values and their effects.  The other three show values as italicized value-names and their effects;  this is consistent with +mkdir+ and +mkdir_p+ above (whose names I have now changed from _integer_ to _mode_).  Stet?

File: lib/fileutils.rb [hunk 7]
```diff
@@ -1294,20 +1301,22 @@ def remove_file(path, force = false)
   end
   module_function :remove_file
 
+  # Recursively removes the directory entry given by +path+,
+  # which should be the entry for a regular file, a symbolic link,
+  # or a directory.
   #
-  # Removes a directory +dir+ and its contents recursively.
-  # This method ignores StandardError if +force+ is true.
+  # Keyword arguments:
```
Comment (peterzhu2118) [inline lib/fileutils.rb:1308]: This also needs to be fixed (not a keyword argument).

Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**